igor_src/main.cpp

```diff
@@ -1876,8 +1908,50 @@ int main(int argc , char* argv[]){
 					return terminate_IGoR_with_error_message("Exception caught upon aligning J genomic templates.",e);
 				}
 			}
+			
+			//Get CDR3 from alignments.
+			if (bFeature_CDR3){
+				unordered_map<int,pair<string,unordered_map<Gene_class,vector<Alignment_data>>>> sorted_alignments;
+				try{
+					sorted_alignments = read_alignments_seq_csv_score_range(cl_path + "aligns/" +  batchname + v_align_filename, V_gene , 55 , false , indexed_seqlist  );
+				}
+				catch(exception& e){
+					return terminate_IGoR_with_error_message("Exception caught while reading V alignments before inference/evaluation. Make sure alignments were carried previously using \"-align --V\" or \"-align --all\" with similar path parameters (working directory, batchname, ...)",e);
+				}
 
-		}
+				try{
+					sorted_alignments = read_alignments_seq_csv_score_range(cl_path + "aligns/" +  batchname + j_align_filename, J_gene , 10 , false , indexed_seqlist , sorted_alignments);
+				}
+				catch(exception& e){
+					return terminate_IGoR_with_error_message("Exception caught while reading J alignments before inference/evaluation. Make sure alignments were carried previously using \"-align --J\" or \"-align --all\" with similar path parameters (working directory, batchname, ...)",e);
+				}
+			
+				string cl_path_ref_genome = string(IGOR_DATA_DIR) + "/models/"+species_str+"/"+chain_path_str+"/ref_genome/";
+				string cl_path_aligns     = cl_path + "aligns/" +batchname;
+
+				string flnV_CDR3_anchors = cl_path_ref_genome + "V_gene_CDR3_anchors.csv";
+				string flnJ_CDR3_anchors = cl_path_ref_genome + "J_gene_CDR3_anchors.csv";
+
+
+				string flnIndexedCDR3      = cl_path_aligns     + "indexed_CDR3.csv";
+				ofstream ofileIndexedCDR3(flnIndexedCDR3);
+				ofileIndexedCDR3 << "seq_index;v_anchor;j_anchor;CDR3nt;CDR3aa"<<endl;
+			
+				ExtractFeatures featureCDR3;
+				featureCDR3.load_VJgenomicTemplates(v_genomic, j_genomic);
+				featureCDR3.load_VJanchors(flnV_CDR3_anchors, flnJ_CDR3_anchors);
+				featureCDR3.set_sorted_alignments(&sorted_alignments);
+				
+				// For each sequence get the CDR3
+				for (auto seq_it = indexed_seqlist.begin(); seq_it != indexed_seqlist.end(); ++seq_it){
```

Comment:
```suggestion
				for (Indexed_Seq indexed_seq : indexed_seqlist){
```
cpp shorthand syntax for a "foreach"